src/io/flutter/editor/ExpressionParsingUtils.java

```diff
@@ -40,4 +40,58 @@ public static Color parseColor(String text, String colorText) {
     //noinspection UseJBColor
     return new Color((int)(value >> 16) & 0xFF, (int)(value >> 8) & 0xFF, (int)value & 0xFF, (int)(value >> 24) & 0xFF);
   }
+
+  public static Color parseColorComponents(String callText, String prefix, boolean isARGB) {
+    if (callText.startsWith(prefix) && callText.endsWith(")")) {
+      final String colorString = callText.substring(prefix.length(), callText.length() - 1).trim();
+      final String[] maybeNumbers = colorString.split(",");
+      if (maybeNumbers.length < 4) {
+        return null;
+      }
+      return isARGB ? parseARGBColorComponents(maybeNumbers) : parseRGBOColorComponents(maybeNumbers);
+    }
+    return null;
+  }
+
+  private static Color parseARGBColorComponents(String[] maybeNumbers) {
+    if (maybeNumbers.length < 4) {
+      return null;
+    }
+    final int[] argb = new int[4];
+    for (int i = 0; i < 4; ++i) {
+      try {
+        argb[i] = Integer.parseUnsignedInt(maybeNumbers[i].trim());
```

Comment:
We're assuming here that the numbers are in decimal; from spot checking fromARGB call sites, is that a reasonable assumption?

Reply:
Thank you, fixed